### ISSUE TITLE
Wrong autocomplete for curly braces

### ISSUE THREAD

**Issue by trogne** [Komodo/KomodoEdit#3464]
### Short Summary
Wrong autocomplete for curly braces.

### Steps to Reproduce
Enter an open curly brace inside the parentheses : 
```
    foo({){
    }
```

### Expected results

```
    foo({}){
    }  
```

### Actual results
The auto-added closing curly brace is wrong : 

```
    foo({){
    }  
    }
```

### Platform Information

*Komodo IDE
*Komodo Version : 11.0.2
*Operating System (and version)?  Windows 10.

**Comment by mitchell-as** [comment]
Hi, what language are you using? I am not able to reproduce this in JavaScript:

1. Start by typing `if () {`.
2. Komodo will autocomplete the closing `}`.
3. Pressing enter yields
```if () {

}
```
4. Place the cursor between the parentheses and type `{`.
5. I do not observe any complement `}` being inserted, even after pressing enter.

**Comment by trogne** [comment]
Javascript.
Me : 
```
if({){
    
}
}
```
Instead of : 
```
if({}){
    
}
```

Do you have this check in preferences, smart editing ? : 
"enable soft characters inside line".

Mine was checked, I unchecked it and I no more have the problem. I guess that's why there's this option.
Maybe not a bug...

**Comment by mitchell-as** [comment]
Thanks for pointing out that preference. If I check that preference, I can reproduce your described issue. I would qualify it as a bug. Thanks!

**Comment by trogne** [comment]
Yes, after thinking, it IS a bug.  I'd prefer to keep the "enable soft characters inside line" active.
